File: Programmers/Lv3/Lv3_기지국설치.py
def solution(n, stations, w):
    answer = 0
    index = 1

    size = 2*w+1
    for i in stations:
        answer += (i-w-index)//size
        if (i-w-index)%size != 0:
            answer += 1
        index = i + w + 1
    if n >= index:
        answer += (n + 1 - index) // size
        if (n + 1 - index) % size != 0:
            answer += 1

    return answer

print(solution(	16, [9], 2))

File: Programmers/Lv3/Lv3_기지국설치.cpp
#include <iostream>
#include <vector>
using namespace std;

int solution(int n, vector<int> stations, int w)
{
	int answer = 0;
	int index = 0;

	for (int i = 1; i <= n; i++) {
		if (index<stations.size() && i >= stations[index] - w) {
			i = stations[index] + w;
			index++;
		}
		else {
			i += (2 * w);
			if (i > n)
				i = n;
			answer++;
		}
	}

	return answer;
}
